aiida_quantumespresso/parsers/projwfc.py

```diff
@@ -347,6 +361,33 @@ def parse(self, **kwargs):
         Dos_out.set_y(dos, 'Dos', 'states/eV')
         self.out('Dos', Dos_out)
 
+    def _parse_xml(self, retrieved_temporary_folder):
+        """Parse the XML file.
+
+        The XML must be parsed in order to obtain the required information for the orbital parsing.
+        """
+        from .parse_xml.exceptions import XMLParseError, XMLUnsupportedFormatError
+        from .parse_xml.pw.parse import parse_xml
+
+        xml_filepath = Path(retrieved_temporary_folder) / self.node.process_class.xml_filename
+
+        if not xml_filepath.exists():
+            self.exit(self.exit_codes.ERROR_OUTPUT_XML_MISSING)
+
+        try:
+            with xml_filepath.open('r') as handle:
+                parsed_xml, logs = parse_xml(handle, None)
+        except IOError:
+            self.exit(self.exit_codes.ERROR_OUTPUT_XML_READ)
```

Comment:
Hmm, I think once `self.exit()` is called, the parsing stops? In the `PwParser` the exit codes are stored in the attributes, and then later in the parsing some logic is applied to determine which exit code should be returned. However, here we _need_ the XML information to parse the outputs, so in case the parsing fails, I think we can just exit with the corresponding exit code.

Reply:
Right, of course, it needs to be returned in the `parse` method. I actually looked at these lines when I saw your comment:

https://github.com/aiidateam/aiida-quantumespresso/blob/27c93e5848829219eda7ab6ee5256bfbbd960f42/aiida_quantumespresso/parsers/pw.py#L109-L118

And somehow failed to notice the `return` statements 🤦 . Thanks for your dilligence @qiaojunfeng! Will fix.